core/src/test/java/org/elasticsearch/index/mapper/core/DateFieldTypeTests.java

```diff
@@ -54,4 +67,49 @@ public void modify(MappedFieldType ft) {
             }
         });
     }
+
+    public void testIsFieldWithinQueryEmptyReader() throws IOException {
+        IndexReader reader = new MultiReader();
+        DateFieldType ft = new DateFieldType();
+        ft.setName("my_date");
+        assertEquals(Relation.DISJOINT, ft.isFieldWithinQuery(reader, "2015-10-12", "2016-04-03",
+                randomBoolean(), randomBoolean(), null, null));
+    }
+
+    public void testIsFieldWithinQuery() throws IOException {
```

Comment:
Should we add a test to each of the other types to check that for random query bounds `MappedFieldType.isFieldWithinQuery(IndexReader, Object, Object, boolean, boolean, TimeZone, DateMathParser)` always  returns Relation.INTERSECTS?